### PR TITLE
docs: Update the roadmap for v1.12

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
Update the roadmap for the 1.12 release in the July timeframe.

Maintainers, please suggest features that are in the works for the near term so we can add them to this list for 1.12.

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by satoru-takeuchi — APPROVED** [review]

File: ROADMAP.md [hunk 1]
```diff
@@ -8,32 +8,32 @@ We hope that the items listed below will inspire further engagement from the com
...
+  * Easy command for deploying a test cluster [#95](https://github.com/rook/kubectl-rook-ceph/issues/95)
   * Recover the CephCluster CR after accidental deletion [#68](https://github.com/rook/kubectl-rook-ceph/issues/68)
   * Collect details to help troubleshoot the csi driver [#69](https://github.com/rook/kubectl-rook-ceph/issues/69)
-  * Restore mon qourum from OSDs after all mons are lost [#7049](https://github.com/rook/rook/issues/7049)
 
 ## Themes
 
 The general areas for improvements include the following, though may not be committed to a release.
 
-* OSD encryption key rotation [#7925](https://github.com/rook/rook/issues/7925)
```
Comment (Madhu-1) [inline ROADMAP.md:36]: Hey @travisn, do you think it's a good idea to take out the `Support for Windows nodes` feature from Themes? We've stopped working on it and we're confident that it won't be included in the upcoming 1.12 release.

File: ROADMAP.md [hunk 2]
```diff
@@ -8,32 +8,32 @@ We hope that the items listed below will inspire further engagement from the com
...
   * Add alpha support for COSI (Container object storage interface) with K8s 1.25 [#7843](https://github.com/rook/rook/issues/7843)
-  * Support the immutable object cache [#11162](https://github.com/rook/rook/issues/11162)
-* Ceph-CSI v3.8
-  * CephFS encryption support [#3460](https://github.com/ceph/ceph-csi/pull/3460)
-* Update the operator sdk for internal CSV generation [#10141](https://github.com/rook/rook/issues/10141)
-* [Krew plugin](https://github.com/rook/kubectl-rook-ceph) features planned in the 1.11 time frame
+  * Pool sharing for clusters where many object stores are required [#11411](https://github.com/rook/rook/issues/11411)
+* Ceph-CSI v3.9
+
+## Krew Plugin
```
Comment (subhamkrai) [inline ROADMAP.md:25]: remove L15 and move to krew section regarding recovering from all mons down using OSD https://github.com/rook/kubectl-rook-ceph/issues/74

**Review by subhamkrai — CHANGES_REQUESTED** [review]

**Review by galexrt — APPROVED** [review]
Besides Subham's comment, LGTM

**Comment by travisn** [comment]
> I would say these are some features we need: IPV6 support for external cluster: #11602
> 
> Reducing a step for installation: #12207
> 
> Helm chart should not upgrade Storage classes: #11398

Thanks, I'll add the ipv6 support to the list. The other two I would consider bug fixes that we don't need to include.

**Review by subhamkrai — APPROVED** [review]
LGTM!

File: ROADMAP.md (hunk 1 above)
Comment (Madhu-1) [reply ROADMAP.md:36]: Yes, but it was started around two years back and is now a stale item for CSI. It might come/might not come to cephCSI.

**Review by parth-gr — APPROVED** [review]

File: ROADMAP.md (hunk 1 above)
Comment (Madhu-1) [reply ROADMAP.md:36]: lets keep it for now and see if we need to remove it later.

**Review by Madhu-1 — APPROVED** [review]

File: ROADMAP.md (hunk 1 above)
Comment (travisn) [reply ROADMAP.md:36]: Right, it's still a future wish, so we can keep it in this section for now

**Pull request merged**